### DISCUSSION TITLE
Bulk data in FDRS

### DISCUSSION THREAD

**Discussion by timmbogner** [timmbogner/Farm-Data-Relay-System#148] [Ideas]
This is a feature that is very far off, but I want to document it here so that I can close the issue that gave me the idea ( #85).

Packets current come in two types: either a single **SystemPacket** that is less than 7 bytes, or one or more **DataReadings** which are 7+ bytes. 

My idea is to implement a third type of packet that is exactly 250 bytes. This packet would be meant to carry binary data,. It would still have a built-in id, but the rest would be binary. There would probably need to be another layer built on the binary data with frame numbers and maybe other stuff.

This also may prove too messy and need to be scrapped.

The end goal is to send a file over FDRS, such as an image from an ESP-Cam.

**Comment by Stips5** [comment]
Any news about this

**Reply by timmbogner** [reply]
I think this concept would be added sometime after I add CBOR functionality discussed in #187.

My loose schedule is to add the ability for nodes to drop into WiFi OTA mode via an ESP-NOW or LoRa command next (not too complicated), then add CBOR functionality (pretty dang complicated). The actual timeline is still ITA (in-the-air:) but I hope to be working on them soon!

This new datatype (described above) would likely use CBOR, so it would make sense to do this soon after that's all in place.

**Reply by timmbogner** [reply]
Thanks for asking, btw! I hadn't really thought about this in a while.
